integration/ztoc_test.go

```diff
@@ -130,6 +164,221 @@ func TestSociZtocList(t *testing.T) {
 		}
 	})
 }
+func TestSociZtocInfo(t *testing.T) {
+	t.Parallel()
+	sh, done := newSnapshotterBaseShell(t)
+	defer done()
+	rebootContainerd(t, sh, "", "")
+
+	testImages := prepareSociIndices(t, sh)
+
+	getFullZtoc := func(sh *dockershell.Shell, ztocPath string) (*ztoc.Ztoc, error) {
+		output := sh.O("cat", ztocPath)
+		reader := bytes.NewReader(output)
+		z, err := ztoc.Unmarshal(reader)
+		return z, err
+	}
+
+	for _, img := range testImages {
+		img := img
+		tests := []struct {
+			name       string
+			ztocDigest string
+			expectErr  bool
+		}{
+			{
+				name:       "Empty ztoc digest",
+				ztocDigest: "",
+				expectErr:  true,
+			},
+			{
+				name:       "Invalid ztoc digest format",
+				ztocDigest: "hello",
+				expectErr:  true,
+			},
+			{
+				name:       "Invalid ztoc digest length",
+				ztocDigest: "sha256:hello",
+				expectErr:  true,
+			},
+			{
+				name:       "Ztoc digest does not exist",
+				ztocDigest: testutil.RandomDigest(),
+				expectErr:  true,
+			},
+			{
+				name:       "Correct ztoc digest",
+				ztocDigest: img.ztocDigests[0],
+				expectErr:  false,
+			},
+		}
+		for _, tt := range tests {
+			t.Run(tt.name, func(t *testing.T) {
+				var zinfo Info
+				output, err := sh.OLog("soci", "ztoc", "info", tt.ztocDigest)
+				if !tt.expectErr {
+					err := json.Unmarshal(output, &zinfo)
+					if err != nil {
+						t.Fatalf("expected Info type got %s", output)
```

Comment:
can you print the `err` as well? (`t.Fatalf("expected Info type got %s: %v", output, err)`)